### ISSUE TITLE
Error creating or updating private endpoint - Operation returned an invalid status code 'Forbidden'

### ISSUE THREAD

**Issue by rloza08** [ansible/ansible_collections_azure#717]
##### SUMMARY
Encountered an error when creating Private Endpoint.

##### ISSUE TYPE
- Bug Report
```
The full traceback is:
  File "/tmp/ansible_azure.azcollection.azure_rm_privateendpoint_payload_sutpvvox/ansible_azure.azcollection.azure_rm_privateendpoint_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_privateendpoint.py", line 312, in create_update_resource_private_endpoint
  File "/var/lib/awx/venv/azcollection-1.10.0/lib/python3.6/site-packages/azure/mgmt/network/v2020_06_01/operations/_private_endpoints_operations.py", line 267, in create_or_update
    **operation_config
  File "/var/lib/awx/venv/azcollection-1.10.0/lib/python3.6/site-packages/azure/mgmt/network/v2020_06_01/operations/_private_endpoints_operations.py", line 220, in _create_or_update_initial
    raise models.ErrorException(self._deserialize, response)
```

##### COMPONENT NAME
azure_rm_privateendpoint

##### ANSIBLE VERSION

```paste below
ansible-playbook 2.9.23
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/var/lib/awx/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 3.6.8 (default, Aug 12 2021, 07:06:15) [GCC 8.4.1 20200928 (Red Hat 8.4.1-1)]
```

##### COLLECTION VERSION

```paste below
v1.10.0
```

##### OS / ENVIRONMENT

RedHat/Ubuntu

##### STEPS TO REPRODUCE

```
---
- name: Private Endpoint
  hosts: localhost
  collections:
    - azure.azcollection
  vars:
    ##-- The following are always required to be passed in, usually via previous jobs or workflow payload.
    az_subscription:
    az_resource_group:
    az_virtual_network:
    az_subnet:
    az_private_endpoint:
    az_private_link_service:
    subscription_guid:
  
  gather_facts: no
  tasks:

  - name: Create Private Endpoint
    azure.azcollection.azure_rm_privateendpoint:
      auth_source: auto
      client_id: "{{ az_client_id }}"
      secret: "{{ az_secret }}"
      tenant: "{{ az_tenant_id }}"
      subscription_id: "{{ subscription_guid }}"
      name: "{{ az_private_endpoint }}"
      resource_group: "{{ az_resource_group }}"
      private_link_service_connections:
        - name: Test_private_link_service
          private_link_service_id: /subscriptions/{{ az_subscription }}/resourceGroups/{{ az_resource_group }}/providers/Microsoft.Network/privateLinkServices/{{ az_private_link_service }}
      subnet:
        id: /subscriptions/{{ az_subscription }}/resourceGroups/{{ az_resource_group }}/providers/Microsoft.Network/virtualNetworks/{{ az_virtual_network }}/subnets/{{ az_subnet }}
      tags:
        key1: value1
        key2: value2
```

##### EXPECTED RESULTS

Private Endpoint is created.

##### ACTUAL RESULTS

```
The full traceback is:
  File "/tmp/ansible_azure.azcollection.azure_rm_privateendpoint_payload_sutpvvox/ansible_azure.azcollection.azure_rm_privateendpoint_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_privateendpoint.py", line 312, in create_update_resource_private_endpoint
  File "/var/lib/awx/venv/azcollection-1.10.0/lib/python3.6/site-packages/azure/mgmt/network/v2020_06_01/operations/_private_endpoints_operations.py", line 267, in create_or_update
    **operation_config
  File "/var/lib/awx/venv/azcollection-1.10.0/lib/python3.6/site-packages/azure/mgmt/network/v2020_06_01/operations/_private_endpoints_operations.py", line 220, in _create_or_update_initial
    raise models.ErrorException(self._deserialize, response)
```

**Comment by adthorn** [comment]
Agreed. We need a resolution. We need updated module that supports sub resources (Such as a Keyvault "Vault" sub resource. The documentation is real shallow for examples.

**Comment by Fred-sun** [comment]
@rloza08 Thank you for submitting this issue, but I have tested it locally and have been unable to replicate the problem you encountered? Could you please help provide your test environment? Thank you very much?

```e.g.
TASK [Create private endpoint] *****************************************************************************************
changed: [localhost]

TASK [Get private endpoint info] ***************************************************************************************
ok: [localhost]

TASK [debug] ***********************************************************************************************************
ok: [localhost] => {
    "output": {
        "changed": false,
        "failed": false,
        "privateendpoints": [
            {
                "etag": "W/\"bfb82d09-f081-4536-997f-99543ce0db02\"",
                "id": "/subscriptions/xxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/privateEndpoints/t",
                "location": "eastus",
                "name": "t",
                "network_interfaces": [
                    "/subscriptions/xxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/networkInterfaces/t.nic.0666aa23-2560-4ae2-bbf3-5a1e4fd48ac3"
                ],
                "private_link_service_connections": [
                    "/subscriptions/xxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/privateEndpoints/t/privateLinkServiceConnections/privateEndpoints_test_name02"
                ],
                "provisioning_state": "Succeeded",
                "subnet_id": "/subscriptions/xxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/virtualNetworks/vnetprivate03/subnets/subnetprivate03",
                "tags": {
                    "key1": "value1",
                    "key2": "value2"
                },
                "type": "Microsoft.Network/privateEndpoints"
            }
        ]
    }
}

**Comment by adthorn** [comment]
RHEL 7.8, Ansible Tower 3.8.4 and Ansible 2.11.5, azcollection 1.11.0. We've updated to a brand new Python 3 virtual environment.

**Comment by adthorn** [comment]
Not for nothing, we're not connecting to a custom private link service, we're connecting to each one of these azure services private link service. 

So you create a key vault or a storage account, we're connecting to that service via a private endpoint.

**Comment by Fred-sun** [comment]
@adthorn Would you please provide the whole execution process and playbook, It will helpfull to resolve the problem! Thank you very much!

**Comment by adthorn** [comment]
This can be resolved by quoting all of your resources @rloza08 .

    - name: Create private endpoint for Keyvault
      azure_rm_privateendpoint:
        name: "{{azr_pve_name}}"
        subscription_id: "{{azr_subid}}"
        tenant: "{{tenant}}"
        secret: "{{secret}}"
        client_id: "{{client_id}}"
        resource_group: "{{azr_rg_name}}"
        subnet:
          id: "/subscriptions/{{azr_subid}}/resourceGroups/{{azr_rg_name}}/providers/Microsoft.Network/virtualNetworks/{{azr_vnet_name}}/subnets/{{azr_snet_sec}}"
        private_link_service_connections:
          - name: "{{azr_pve_name}}"
            private_link_service_id: "{{keyvault_output.id}}"
            group_ids: "{{azr_pve_sub_resource}}"
        tags:
            Description: "Private Endpoint: For Resource {{azr_kv_name}}"
            Subscription: "{{subscode}}"
            AppCode: "{{appcode}}"
            AppOwner: "{{appowner}}"
            Environment: "{{envcode}}"
            Location: "{{location}}us"
            Instance: "{{pve_kv_instance}}"
            ResourceType: "pve"
      debugger: on_failed
      register: pve_output

**Comment by rloza08** [comment]
Thanks! This has been resolved by upgrading to azcollection 1.11.0
